test_simple_sb/test_simple_sb_magma_no_model.py

```diff
@@ -0,0 +1,175 @@
+import tempfile
+import random
+import magma
+import fault
+import fault.random
+from common.core import Core
+from common.testers import BasicTester
+from simple_sb.simple_sb_magma import SB
+from simple_sb.simple_sb_model import SideModel
+
+
+class DummyCore(Core):
+    def __init__(self):
+        super().__init__()
+
+        TData = magma.Bits(16)
+        TBit = magma.Bits(1)
+
+        self.add_ports(
+            data_in=magma.In(TData),
+            bit_in=magma.In(TBit),
+            data_out=magma.Out(TData),
+            bit_out=magma.Out(TBit),
+        )
+
+        self.wire(self.ports.data_in, self.ports.data_out)
+        self.wire(self.ports.bit_in, self.ports.bit_out)
+
+    def inputs(self):
+        return (self.ports.data_in, self.ports.bit_in)
+
+    def outputs(self):
+        return (self.ports.data_out, self.ports.bit_out)
+
+    def name(self):
+        return "DummyCore"
+
+
+def test_regression():
+    # Create magma circuit.
+    dummy_core = DummyCore()
+    simple_sb = SB(dummy_core.outputs())
+
+    for output in dummy_core.outputs():
+        port_name = output._name
+        port = simple_sb.ports.get(port_name, None)
+        assert port is not None
+        assert port.type() == output.type().flip()
+
+    NUM_TRACKS = 5
+    LAYERS = (1, 16)
+    SIDES = ("north", "west", "south", "east")
+    # num_core_outputs = len(dummy_core.outputs())
+
+    simple_sb_circuit = simple_sb.circuit()
+    # tester = fault.Tester(simple_sb_circuit, simple_sb_circuit.clk)
```

Comment:
should probably get rid of these comments since you're using the BasicTester instead

Reply:
Removed, thanks for pointing that out!